File: Solutions in Python/p024.py
# 
# Solution to Project Euler problem 24
# Copyright (c) Sankha Saha (CODR). All rights reserved.
# 
#
# https://github.com/codr07/Project-Euler-Solusions
# 

import itertools


# We initialize a list as the lowest permutation of the given digits, which is the sequence
# (0,1,2,3,4,5,6,7,8,9). Then we call a Python library function that generates a stream of
# all permutations of the values, seek to the 999 999th element (0-based indexing), and stringify it.
def compute():
	arr = list(range(10))
	temp = itertools.islice(itertools.permutations(arr), 999999, None)
	return "".join(str(x) for x in next(temp))


if __name__ == "__main__":
	print(compute())

File: codes/mathematica/p024.m
(* 
 * Solution to Project Euler problem 24
 * Copyright (c) Sankha Saha. All rights reserved.
 * 
 *  
 *   
 *)


(* 
 * We generate all 10! permutations of the sequence (0,1,2,3,4,5,6,7,8,9)
 * into memory, and select the 1 000 000th element (1-based indexing).
 *)
FromDigits[Permutations[Range[0, 9]][[1000000]]]
